### ISSUE TITLE
Unwrapping errors.Join error to []error through type assertion.

### ISSUE THREAD

**Issue by forgoty** [polyfloyd/go-errorlint#61]
I am trying to type assert multiples errors created with `errors.Join`

```go
err := errors.Join(errors.New("err1"), errors.New("err2"))
errs, ok := err.(interface{ Unwrap() []error })
```
This code results in a warning:
```
type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors (errorlint)
        errs, ok := err.(interface{ Unwrap() []error })
                         ^
```

This is a valid case for me from go 1.20. Seems should be whitelisted.

**Comment by polyfloyd** [comment]
Fixed! Thanks for reporting :)
